include/drivers/dai.h

```diff
@@ -0,0 +1,717 @@
+	 *
+	 * The interface was initialized but is not yet ready to receive /
+	 * transmit data. Call dai_configure() to configure interface and change
+	 * its state to READY.
+	 */
+	DAI_STATE_NOT_READY = 0,
+	/** The interface is ready to receive / transmit data. */
+	DAI_STATE_READY,
+	/** The interface is receiving / transmitting data. */
+	DAI_STATE_RUNNING,
+	/** The interface is clocking but not receiving / transmitting data. */
+	DAI_STATE_PRE_RUNNING,
+	/** The interface is draining its transmit queue. */
+	DAI_STATE_STOPPING,
+	/** TX buffer underrun or RX buffer overrun has occurred. */
+	DAI_STATE_ERROR,
+	/** TBD not used today, here for completeness */
+	DAI_STATE_DRAINING,
+};
+
+/** Trigger command */
+enum dai_trigger_cmd {
+	/** @brief Start the transmission / reception of data.
+	 *
+	 * If DAI_DIR_TX is set some data has to be queued for transmission by
+	 * the dai_write() function. This trigger can be used in READY state
+	 * only and changes the interface state to RUNNING.
+	 */
+	DAI_TRIGGER_START = 0,
+	/** @brief Optional - Pre Start the transmission / reception of data.
+	 *
+	 * Allows the DAI and downstream codecs to prepare for audio Tx/Rx by
+	 * starting any required clocks for downstream PLL/FLL locking.
+	 */
+	DAI_TRIGGER_PRE_START,
+	/** @brief Stop the transmission / reception of data.
+	 *
+	 * Stop the transmission / reception of data at the end of the current
+	 * memory block. This trigger can be used in RUNNING state only and at
+	 * first changes the interface state to STOPPING. When the current TX /
+	 * RX block is transmitted / received the state is changed to READY.
+	 * Subsequent START trigger will resume transmission / reception where
+	 * it stopped.
+	 */
+	DAI_TRIGGER_STOP,
+	/** @brief Optional - Post Stop the transmission / reception of data.
+	 *
+	 * Allows the DAI and downstream codecs to shutdown cleanly after audio
+	 * Tx/Rx by stopping any required clocks for downstream audio completion.
+	 */
+	DAI_TRIGGER_POST_STOP,
+	/** @brief Empty the transmit queue.
+	 *
+	 * Send all data in the transmit queue and stop the transmission.
+	 * If the trigger is applied to the RX queue it has the same effect as
+	 * DAI_TRIGGER_STOP. This trigger can be used in RUNNING state only and
+	 * at first changes the interface state to STOPPING. When all TX blocks
+	 * are transmitted the state is changed to READY.
+	 */
+	DAI_TRIGGER_DRAIN,
+	/** @brief Discard the transmit / receive queue.
+	 *
+	 * Stop the transmission / reception immediately and discard the
+	 * contents of the respective queue. This trigger can be used in any
+	 * state other than NOT_READY and changes the interface state to READY.
+	 */
+	DAI_TRIGGER_DROP,
+	/** @brief Prepare the queues after underrun/overrun error has occurred.
+	 *
+	 * This trigger can be used in ERROR state only and changes the
+	 * interface state to READY.
+	 */
+	DAI_TRIGGER_PREPARE,
+	/** @brief Reset
+	 *
+	 * This trigger frees resources.
```

Comment:
back to the initial DAI state on driver init.

Reply:
ok